src/modules/Game.class.js

```diff
@@ -1,68 +1,301 @@
 'use strict';
 
-/**
- * This class represents the game.
- * Now it has a basic structure, that is needed for testing.
- * Feel free to add more props and methods if needed.
- */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
+  static STATUS = {
+    IDLE: 'idle',
+    PLAYING: 'playing',
+    WIN: 'win',
+    LOSE: 'lose',
+  };
+
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
     // eslint-disable-next-line no-console
-    console.log(initialState);
+    this.status = Game.STATUS.IDLE;
+    this.initialState = initialState;
+    this.state = initialState.map((row) => [...row]);
+    this.score = 0;
+  }
+
+  getScore() {
+    return this.score;
+  }
+
+  getState() {
+    return this.state;
+  }
+
+  getStatus() {
+    return this.status;
+  }
+
+  start() {
+    this.status = Game.STATUS.PLAYING;
+    this.state = this.initialState.map((row) => [...row]);
+    this.addNewNumber();
+    this.addNewNumber();
+  }
+
+  restart() {
+    this.status = Game.STATUS.IDLE;
+    this.state = this.initialState.map((row) => [...row]);
+    this.score = 0;
+  }
+
+  addNewNumber() {
+    const emptyPositions = this.getEmptyPosition();
+
+    if (emptyPositions.length === 0) {
+      return;
+    }
+
+    const randomIndex = Math.floor(Math.random() * emptyPositions.length);
+    const [row, col] = emptyPositions[randomIndex];
+
+    this.state[row][col] = Math.random() < 0.9 ? 2 : 4;
```

Comment:
The random number generation logic is a key part of the game's mechanics. It might be beneficial to isolate this logic into a separate method for better testability and potential adjustments in the future.